rusk-wallet/src/lib/gql.rs

```diff
@@ -0,0 +1,150 @@
+    pub fn tx_status(&self, tx_id: &str) -> Result<TxStatus, GraphQLError> {
+        // graphql connection
+        let client = Client::new(&self.url);
+
+        // helper structs to deserialize response
+        #[derive(Deserialize)]
+        struct Tx {
+            pub txerror: String,
+        }
+        #[derive(Deserialize)]
+        struct Transactions {
+            pub transactions: Vec<Tx>,
+        }
+
+        let query =
+            "{transactions(txid:\"####\"){ txerror }}".replace("####", tx_id);
+
+        let response = block_in_place(move || {
+            Handle::current().block_on(async move {
+                client.query::<Transactions>(&query).await
+            })
+        });
+
+        match response {
+            Ok(data) => match data {
+                Some(d) => {
+                    if !d.transactions.is_empty() {
+                        match d.transactions[0].txerror.is_empty() {
+                            true => Ok(TxStatus::Ok),
+                            false => {
+                                let err_str =
+                                    d.transactions[0].txerror.as_str();
+                                let tx_err =
+                                    serde_json::from_str::<Value>(err_str);
+                                match tx_err {
+                                    Ok(data) => match data["data"].as_str() {
+                                        Some(msg) => {
+                                            Ok(TxStatus::Error(msg.to_string()))
+                                        }
+                                        None => Ok(TxStatus::Error(
+                                            err_str.to_string(),
+                                        )),
+                                    },
+                                    Err(err) => {
+                                        Ok(TxStatus::Error(err.to_string()))
+                                    }
+                                }
+                            }
+                        }
+                    } else {
+                        Ok(TxStatus::NotFound)
+                    }
+                }
+                None => Err(GraphQLError::TxStatus),
+            },
+            Err(err) => {
+                match err.json() {
+                    Some(json) => {
+                        // we stringify the json and use String.contains()
+                        // because GraphQLErrorMessage fields are private
+                        let json_str = format!("{:?}", json[0]);
+                        if json_str.contains("database: transaction not found")
+                        {
+                            Ok(TxStatus::NotFound)
+                        } else {
+                            Err(GraphQLError::Generic(err))
+                        }
+                    }
+                    None => Err(GraphQLError::Generic(err)),
+                }
+            }
+        }
```

Comment:
Consider refactoring this part for better readability. We should avoid using multiple nested blocks - instead, we should perform one check per time, in separated statements